src/test/java/co/rsk/federate/watcher/FederationWatcherTest.java

```diff
@@ -79,248 +76,317 @@ void whenFederationWatcherIsSetUp_shouldAddListener() throws Exception {
-            }
-
-            @Override
-            public void onRetiringFederationChange(Federation newFederation) {
-                retiringCalls.incrementAndGet();
-            }
-        });
         rskListener.onBestBlock(null, null);
 
         // Assert
-        assertEquals(1, activeCalls.get());
-        assertEquals(0, retiringCalls.get());
+        verify(federationProvider).getProposedFederationAddress();
+        verify(federationProvider, never()).getProposedFederation();
+        verify(federationWatcherListener, never()).onProposedFederationChange(any(Federation.class));
+
         verify(federationProvider).getActiveFederationAddress();
         verify(federationProvider).getRetiringFederationAddress();
+        verify(federationWatcherListener).onActiveFederationChange(FIRST_FEDERATION);
+
         verify(federationProvider).getActiveFederation();
         verify(federationProvider, never()).getRetiringFederation();
+        verify(federationWatcherListener, never()).onRetiringFederationChange(any(Federation.class));
     }
 
     @Test
-    void whenNoActiveFederationChange_shouldNotTriggerActiveOrRetiringFederationChange() throws Exception {
+    void onBestBlock_whenActiveFederationChangesFromActiveToOtherActive_shouldTriggerActiveFederationChange() throws Exception {
         // Arrange
-        var rskListener = setupAndGetRskListener(FIRST_FEDERATION, null);
-        var activeCalls = new AtomicInteger(0);
-        var retiringCalls = new AtomicInteger(0);
+        var rskListener = setupAndGetRskListener(FIRST_FEDERATION, null, null);
+        when(federationProvider.getProposedFederationAddress()).thenReturn(Optional.empty());
+        when(federationProvider.getActiveFederationAddress()).thenReturn(SECOND_FEDERATION.getAddress());
+        when(federationProvider.getActiveFederation()).thenReturn(SECOND_FEDERATION);
+        when(federationProvider.getRetiringFederationAddress()).thenReturn(Optional.empty());
+
+        federationWatcher.start(federationProvider, federationWatcherListener);
+
+        // Act
+        rskListener.onBestBlock(null, null);
 
+        // Assert
+        verify(federationProvider).getProposedFederationAddress();
+        verify(federationProvider, never()).getProposedFederation();
+        verify(federationWatcherListener, never()).onProposedFederationChange(any(Federation.class));
+
+        verify(federationProvider).getActiveFederationAddress();
+        verify(federationProvider).getActiveFederation();
+        verify(federationWatcherListener).onActiveFederationChange(SECOND_FEDERATION);
+
+        verify(federationProvider).getRetiringFederationAddress();
+        verify(federationProvider, never()).getRetiringFederation();
+        verify(federationWatcherListener, never()).onRetiringFederationChange(any(Federation.class));
+    }
+
+    @Test
+    void onBestBlock_whenNoActiveFederationChange_shouldNotTriggerActiveOrRetiringFederationChange() throws Exception {
```

Comment:
```suggestion
    void onBestBlock_whenNoActiveFederationChange_shouldNotTriggerAnyFederationChange() throws Exception {
```
since you are also checking for the proposed right?

Reply:
Updated